<launch>	
<!--
    <include file="$(find openni2_launch)/launch/openni2.launch" />
	<node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" respawn="true" output="screen" />
-->
	<include file="$(find sara_moveit)/launch/wm_move_group.launch" />

	<node name="wm_arm_driver_node" pkg="wm_arm_driver" type="wm_arm_driver_node" output="screen" >
		<remap from="joint_states" to="arm_base_joint_states" />

		<param name="urdf_param" value="/sara" />
		<param name="base_move_group_name" value="base" />
		<param name="manipulator_move_group_name" value="manipulator" />

		<!-- max_velocity in degrees/second -->
		<param name="max_velocity_joints123" value="20" />
		<param name="max_velocity_joints456" value="20" />
		<param name="joint_movement_synchronization" value="true" />

		<!-- home_position and joints_offset must be in degrees -->
		<param name="home_position" value="[0.0, 0.0, 0.0, 0.0, 0.0]" />
		<param name="joints_offset" value='[166.60, 89.25, 24.26, 350.85, 331.23, 0.0]' />		

		<param name="max_torque_joint1" value="20" />
		<param name="max_torque_joint2" value="20" />
		<param name="max_torque_joint3" value="20" />
		<param name="max_torque_joint4" value="20" />
		<param name="max_torque_joint5" value="20" />
		<param name="max_torque_joint6" value="20" />

		<!-- callbacks rate, in Hertz -->
		<param name="torque_callback_rate" value="20" />
		<param name="position_callback_rate" value="20" />
		<param name="velocity_callback_rate" value="20" />

		<!-- planner parameters -->
		<param name="planning_time" value="12" />
		<param name="planning_attempts" value="4" />
		<param name="goal_position_tolerance" value="0.01" />
		<param name="goal_orientation_tolerance" value="0.4" />
		<param name="base_planner" value="RRTConnectkConfigDefault" />
		<param name="manipulator_planner" value="RRTConnectkConfigDefault" />

		<param name="manipulator/kinematics_solver_timeout" value="0.1" />
		<param name="manipulator/solve_type" value="Speed" />

		<!-- action server parameters -->
		<param name="action_server_name" value="execute_plan" />
		<param name="action_feedback_rate" value="5" />
		<param name="wait_duration" value="0.5" />
		<param name="maximum_skip" value="2" />

	</node>
</launch>
